service/src/main/java/uk/nhs/adaptors/gp2gp/common/storageconnectors/BlobContainerClientConfig.java

```diff
@@ -0,0 +1,35 @@
+package uk.nhs.adaptors.gp2gp.common.storageconnectors;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import com.azure.storage.blob.BlobContainerClient;
+import com.azure.storage.blob.BlobServiceClient;
+import com.azure.storage.blob.BlobServiceClientBuilder;
+import static uk.nhs.adaptors.gp2gp.common.storageconnectors.StorageConnectorOptions.AZURE;
+
+@Configuration
+@ComponentScan(basePackageClasses = StorageConnectorFactory.class)
```

Comment:
What is the ComponentScan doing here. Is it needed? Spring Boot scan's the application's package and sub-packages already.